device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/Socks5SocketFactory.java

```diff
@@ -181,6 +183,7 @@ public int getLocalPort() {
             return isLocal() ? super.getLocalPort() : mProxied.mPort;
         }
 
+        @SneakyThrows
```

Comment:
I don't think this is the correct idea. This changes the behavior of the method to throw the exception rather than just return null when an UnknownHostException is thrown. I'd recommend logging the exception rather than doing a sneaky throws

Reply:
I am not sure if just logging will get rid of this warning. I will check that.

Reply:
Does work. I have updated to log it instead.